### DISCUSSION TITLE
Is there a timeframe for the release of Version1.0.0?

### DISCUSSION THREAD

**Discussion by BicanGuel** [argoproj-labs/argocd-image-updater#857] [Q&A] [answered]
We want to use the argocd-image-updater for non-productive workloads and are trying to get a feeling when we can actually use it for production.

Answers like: within 3 months OR unknown timeframe; is totally fine. We just want to get a grasp.

Thanks in advance.

**Comment by jannfis** [comment, marked as answer]
TL;DR: Sorry, we do not have a schedule for v1.0 yet.

Longer version:

The project was unfortunately hibernating for a while (lack of maintainers, lack of time) but we recently picked up some pace again. So bugs will be fixed and new features be incorporated.

There's still some large chunks of work and breaking changes ahead of us for a v1.0 version, due to the attached promises of semver. At some point in time, and this is the largest chunk before a v1.0, the way to configure image updater will eventually change. So what is annotations now will likely become its own CRD/Kubernetes API. That being said, a lot of people actually use image updater already for production purposes. However, breaking changes between minor releases must be expected.

**Reply by BicanGuel** [reply]
Ok thanks for the update.

**Reply by RmStorm** [reply]
Is there a design doc somewhere to get an idea about the design for a 1.0 version? The annotation based approach works very well in practice!

And as you say yourself a lot of people are already running this in production! Based on [this survey (N=123)](https://blog.argoproj.io/cncf-argo-cd-rollouts-2023-user-survey-results-514aa21c21df) 15% of ArgoCD users use it. Sadly it doesn't tell us which percentage run it in prod :sweat_smile:.
